### PR TITLE
ARROW-5168: [GLib] Add garrow_array_take()

### PR DESCRIPTION
**Pull request by shiro615**

### CODE REVIEW

File: c_glib/test/test-take.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+class TestTake < Test::Unit::TestCase
+  include Helper::Buildable
+  include Helper::Omittable
+
+  def test_default
+    require_gi(1, 42, 0)
```
Comment (kou) [inline c_glib/test/test-take.rb]: We don't need this when we don't have nullable argument.

File: c_glib/test/test-take.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+class TestTake < Test::Unit::TestCase
+  include Helper::Buildable
+  include Helper::Omittable
+
+  def test_default
```
Comment (kou) [inline c_glib/test/test-take.rb]: Could you use `test_no_null` or something?

File: c_glib/test/test-take.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    require_gi(1, 42, 0)
+    indices = build_int16_array([1, 0, 2])
+    assert_equal(build_int16_array([0, 1, 2]),
+                 build_int16_array([1, 0 ,2]).take(indices))
+  end
```
Comment (kou) [inline c_glib/test/test-take.rb]: Could you also add a test for indices with NULL such as `build_int16_array([2, nil, 0])` case?

File: c_glib/arrow-glib/compute.cpp [hunk 4]
```diff
@@ -902,6 +902,43 @@ garrow_double_array_sum(GArrowDoubleArray *array,
...
+ * @error: (nullable): Return location for a #GError or %NULL.
+ *
+ * Returns: (nullable) (transfer full): The #GArrowArray taken from
+ *   an array of values at indices in input array or %NULL on error.
+ *
+ * Since: 0.14.0
+ */
+GArrowArray *
+garrow_array_take(GArrowArray *array,
+                  GArrowArray *indices,
```
Comment (kou) [inline c_glib/arrow-glib/compute.cpp]: Could you add `GArrowTakeOptions *options` to avoid backward incompatibility in the future?

**Review by kou — CHANGES_REQUESTED** [review]
Thanks! It almost good.
Could you check my comments?

File: c_glib/test/test-take.rb (hunk 1 above)
Comment (shiro615) [reply c_glib/test/test-take.rb]: I see. I left this to add `GArrowTakeOptions *options`.

File: c_glib/arrow-glib/compute.cpp (hunk 4 above)
Comment (shiro615) [reply c_glib/arrow-glib/compute.cpp]: I've added it.

File: c_glib/test/test-take.rb (hunk 2 above)
Comment (shiro615) [reply c_glib/test/test-take.rb]: I've changed `test_default` to `test_no_null`.

File: c_glib/test/test-take.rb (hunk 3 above)
Comment (shiro615) [reply c_glib/test/test-take.rb]: I've added `test_null`.

**Review by kou — APPROVED** [review]
+1

**Pull request closed without merging**